app/src/main/java/org/oppia/app/player/state/StateFragment.kt

```diff
@@ -52,12 +65,12 @@ class StateFragment : InjectableFragment(), CellularDataInterface {
 
   override fun enableAudioWhileOnCellular(saveUserChoice: Boolean) {
     stateFragmentPresenter.setAudioFragmentVisible(true)
-    // saveUserChoice -> true -> save this preference
-    // saveUserChoice -> false -> do not save this preference
+    if (saveUserChoice) cellularDialogController.setHideDialogPreference(true)
+    cellularDialogController.setUseCellularDataPreference(true)
```

Comment:
Shouldn't we only be changing this value permanently if the user asks to not show the dialog again? Otherwise we should always re-ask.

Similarly, shouldn't we be setting useCellularData here, and only setting useCellularData in the observer callback above if we shouldn't show the cellular dialog again?